### PR TITLE
Made collection edit redirect to root collection when passed an invalid id

### PR DESCRIPTION
**Pull request by marcelmoosbrugger**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | -
| Related issues/PRs | -
| License | MIT
| Documentation PR | -

#### What's in this PR?

When the collection-edit component is passed an invalid or not existent id, for which the server responds with an error, the collection-edit then navigates to the root collection.

#### Why?

Ids of not existent collections got passed to the collection-edit when a user's last visited collection got deleted by another user. If now the first user navigated back to the media bundle, Sulu would try to navigate to the deleted collection.

### CODE REVIEW

**Comment by marcelmoosbrugger** [comment]
@danrot This PR can be reviewed. FYI it's not dependent on any other open pull request.

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/components/collections/edit/main.js [hunk 1]
```diff
@@ -131,14 +131,28 @@ define([
...
             if (!!this.options.id) {
-                return CollectionManager.load(this.options.id, this.options.locale);
+
+                CollectionManager.load(this.options.id, this.options.locale)
+                    .then(function(data) {
+                        whenDataLoaded.resolve(data);
+                    })
+                    .fail(function() {
+                        UserSettingsManager.setLastVisitedCollection(null);
+                        MediaRouter.toRootCollection();
```
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/components/collections/edit/main.js]: Why is the `whenDataLoaded` deferred not resolved here?

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/services/media-router.js [hunk 2]
```diff
@@ -50,15 +43,60 @@ define([
         toRootCollection: function(locale) {
             locale = locale || UserSettingsManager.getMediaLocale();
             Mediator.emit('sulu.router.navigate', 'media/collections/' + locale, true, true);
```
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/services/media-router.js]: Isn't the advantage of this pattern that you don't have to hardcode the url here?

File: src/Sulu/Bundle/MediaBundle/Resources/public/js/components/collections/edit/main.js [hunk 3]
```diff
@@ -131,14 +131,29 @@ define([
          * @returns {*}
          */
         loadComponentData: function() {
+            var whenDataLoaded = $.Deferred();
+
             if (!!this.options.id) {
-                return CollectionManager.load(this.options.id, this.options.locale);
+
```
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Resources/public/js/components/collections/edit/main.js]: Please remove that new line

**Pull request merged**